### ISSUE TITLE
Implement OSProcess#toString()

### ISSUE THREAD

**Issue by retomerz** [oshi/oshi#607]
Please implement OSProcess#toString() like this:
```
    @Override
    public String toString() {
      return "<" + this.processID + "> " + this.name;
    }
```
This can help while debugging.

**Comment by dbwiddis** [comment]
Sounds like a great idea!  Please submit a PR!

**Comment by dbwiddis** [comment]
Thinking about this for a bit, I'm wondering if we should limit it to just those fields.   It might be nice to have a standard `toString()` for all OSHI output -- if we used JSON as the `toString()` we could even get rid of the entire `oshi-json` artifact.   It looks like there are some "standards" in packages as well, like [this Apache Commons option](https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/builder/ToStringStyle.html#DEFAULT_STYLE) (perhaps with a configurable style we could set at the project level).

Overkill?  Perhaps now, but I expect this will be re-thought with version 4.0.

**Comment by cilki** [comment]
Having `toString()` return a huge JSON string is probably not convenient for debugging purposes. A universal `toJson()` may be a good idea unless it would require an unnecessary dependency.

In this case, I agree with including the PID and name of a process in its String representation.

**Comment by retomerz** [comment]
@dbwiddis Keep it simple. Add Apache Commons dependency for this is just overkill.
And every dependency is a counterargument to use a library because of JAR version conflicts
(most enterprise applications still using Java 8. And not 9, or newer, with the Java module system).
And a JSON API should look different. Using toString() for JSON is inefficient because such a API would create tons of String and StringBuilder instances at runtime. toString() is for the programmer.
The right way, in case of JSON support, is to pass a "JSON Writer" to the object, or pass the object to a writer, so the writer can stream the data.
Back to the toString:
Personally I prefer to include the instance id (hashCode) in the toString so you can easily see in the debugger if two instances are really the same or not.
So this would be my preferred implementation: 
```
  @Override
  public String toString() {
    return "OSProcess@" + Integer.toHexString(hashCode()) + "{" +
            "pid=" + this.pid +
            ", name='" + this.name + '\'' +
            '}';
  }
```

**Comment by dbwiddis** [comment]
I agree "keep it simple" and that there should be a separate JSON output (if we go there, although I'd lean toward just having a layer of serializable objects that users can manipulate with their choice of third party libraries like Jackson, and leave the JSON choices to the user).  Again, that's all future thinking.  

I didn't intend to suggest adding Apache Commons as a dependency, merely to point out that their format is somewhat more of a "standard" than arbitrary angle brackets.   I agree with your suggestion to follow that "default" format with an abbreviated list of elements that we care about (pid and name), so `OSProcess@182f0db[pid=124,name=oshi]` would be better, IMO than `<124> oshi`.

**Comment by retomerz** [comment]
> OSProcess@182f0db[pid=124,name=oshi]

Looks perfect for me.
